### PR TITLE
Release notes + fixes for header levels + template

### PR DESCRIPTION
**Pull request by artemkovalyov**
## What Has Changed?

Explain what you have changed.

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on it's own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)

### CODE REVIEW

File: docs/java/release-notes.mdx [hunk 1]
```diff
@@ -31,22 +31,67 @@ We highly recommend regularly updating to the latest SDK version. It will help y
...
+### Improvements
+
+- Dependency Updates:
+  - SAP dependency updates:
+    - Update [SAP Cloud Security Client](https://github.com/SAP/cloud-security-xsuaa-integration) from `2.7.6` to `2.8.0`
+    - Update API of Java EE 7 Web Profile TomEE 7 from `1.87.4` to `1.88.6`
+    - Update API of Java Web Tomcat 8 from `3.118.5` to `3.119.9`
+  - Other dependency updates:
+    - Major version updates:
+        - ---Other dependency Major version updates here---
```
Comment (MatKuhr) [inline docs/java/release-notes.mdx:67]: The template stuff needs to be removed

File: docs/java/release-notes.mdx [hunk 2]
```diff
@@ -31,22 +31,67 @@ We highly recommend regularly updating to the latest SDK version. It will help y
...
+### Fixed issues
+
+- We fixed an issue where OData errors originating in OData batch change sets requests could not be parsed correctly.
+
 ## 3.34.1 - December 9, 2020
 
 [Maven Central](https://search.maven.org/search?q=g:com.sap.cloud.sdk*%20AND%20v:3.34.1)
 
-## Compatibility Notes
+### Compatibility Notes
```
Comment (MatKuhr) [inline docs/java/release-notes.mdx:83]: I assume this fixes the table of contents?

Comment (artemkovalyov) [reply docs/java/release-notes.mdx:83]: It fixes the right navigation sidebar if you meant that, @MatKuhr

File: docs/java/release-notes.mdx (hunk 1 above)
Comment (artemkovalyov) [reply docs/java/release-notes.mdx]: Done.

**Review by MatKuhr — APPROVED** [review]

**Pull request merged**